src/utils/fees.ts

```diff
@@ -5,25 +5,26 @@ import {
   Transaction,
   TokenBurnV1,
 } from '@helium/transactions'
-import Balance, {
-  DataCredits,
-  CurrencyType,
-  NetworkTokens,
-} from '@helium/currency'
+import Balance, { DataCredits, CurrencyType } from '@helium/currency'
 import { minBy } from 'lodash'
+import { useSelector } from 'react-redux'
 import { getKeypair } from './secureAccount'
-import { getCurrentOraclePrice, getPredictedOraclePrice } from './appDataClient'
+import { RootState } from '../store/rootReducer'
 
 export const stakingFee = Transaction.stakingFeeTxnAddGatewayV1
 
-export const convertFeeToNetworkTokens = async (
-  balance: Balance<DataCredits>,
-): Promise<Balance<NetworkTokens>> => {
-  const currentPrice = await getCurrentOraclePrice()
-  const predictedPrices = await getPredictedOraclePrice()
-  const prices = [currentPrice, ...predictedPrices]
-  const oraclePrice = minBy(prices, (p) => p.price.integerBalance)
-  return balance.toNetworkTokens(oraclePrice?.price)
+export const useFees = () => {
+  const {
+    heliumData: { currentOraclePrice, predictedOraclePrices },
+  } = useSelector((state: RootState) => state)
+
+  const feeToHNT = (balance: Balance<DataCredits>) => {
+    const prices = [currentOraclePrice, ...predictedOraclePrices]
```

Comment:
Old transactions before data credits were around are crashing here.

I think we need to add `if (!balance) return new Balance<DataCredits>(0, CurrencyType.dataCredit)` for a safeguard